### PR TITLE
Add chunk_opts and fix file path kerfuffle

### LINKED ISSUE #11: issue resolving relative paths
I figured that I would test this on r-novice-inflammation and realized that I should be writing down the barriers that I notice. One of the barriers I'm noticing is that there appears to be a difference in how paths are being handled in the engine. This issue assumes that the documents live in `_episodes_rmd` and are written to a sister folder, `_episodes` with figures being deposited in `figs/`

## Paths

1. The paths in [bin/generate_md_episodes.R](https://github.com/carpentries/styles/blob/6f3004391a9e101285deef8879eb8ae6596a6a92/bin/generate_md_episodes.R#L39-L40) are relative, but it's messing up the rendering of the chunks within the document. I get the feeling that the engine does not inherit chunk options from the parent document even though it is evaluated in the same environment. This can be fixed by setting an absolute path in knit
2. Even if 1 is fixed, there is an issue with the output of the figures. where they are being written in the `_episodes_rmd` folder.

### PR DESCRIPTION
**Pull request by zkamvar**
## Summary

With this PR, it's possible to update the lessons like so (assuming we are in a lesson repo):

```r
library("dovetail")

# Install {pegboard} (what I renamed up2code) 
# and mass-convert the episodes in the lesson to use {dovetail}
source(system.file("convert", "convert.R", package = "dovetail"), echo = TRUE)
```

After this, it should just be a matter of running `make lesson-md` to get this thing going. 

## Detailed changes

 1. I added `dvt_opts()` as a drop-in replacement for [../bin/chunk-options.R](https://github.com/carpentries/styles/blob/2f07ce17b6e32174b689deac3fe1e6658499a4d4/bin/chunk-options.R). This will replace the preamble to the Rmarkdown documents like so:
    ```diff
    -source('../bin/chunk-options.R')
    +library("dovetail")
    +source(dvt_opts())
    ```
2. The engine now sets itself to the base dir if it's specified. I've tested it on the r-novice-inflammation lesson and it appears to be working smoothly (though I will double check that it can indeed read files)
3. inst/convert/convert.R reads in a lesson directory and converts the files to work with dovetail.

This will fix #11

### CODE REVIEW

**Comment by fmichonneau** [comment]
this seems excellent!
also with a new template, we might not need to be quite as constrained as the current repository organization. This repository organization was designed without Rmd files in mind and to some extent having a single folder where all the source lesson files could live could make things easier. It'd be also worth thinking of using the R package structure to make use of the "lessons as packages" paradigm.

File: inst/extdata/produces-plot.Rmd [hunk 1]
```diff
@@ -2,12 +2,13 @@
 title: "produces-image.Rmd"
 author: "Zhian N. Kamvar"
 date: "6/11/2020"
-output: html_document
 ---
 
 ```{r setup, include=FALSE}
 knitr::opts_chunk$set(echo = TRUE)
 library("dovetail")
+source(dvt_opts())
```
[the remark this answers is not in the document]
Comment (zkamvar) [reply inst/extdata/produces-plot.Rmd:10]: dvt_opts() returns the path to [inst/chunk-options.R](https://github.com/carpentries/dovetail/pull/12/files#diff-482d4820dbb1c19312a1ff6f46589a90), so source is sourcing that file.

I was debating whether or not to have `dvt_opts()` set the knitr options outright, but for some reason, I was thinking the relative paths wouldn't be resolved properly (but I'm now realizing that they probably will).

[the remark this answers is not in the document]
Comment (zkamvar) [reply inst/extdata/produces-plot.Rmd:10]: Also: this is my first attempt at getting an consistent naming convention (something that I'm not terribly great at), so any critique is very welcome.

**Comment by zkamvar** [comment]
> also with a new template, we might not need to be quite as constrained as the current repository organization. This repository organization was designed without Rmd files in mind and to some extent having a single folder where all the source lesson files could live could make things easier. It'd be also worth thinking of using the R package structure to make use of the "lessons as packages" paradigm.

*For my own reference, the Lessons as packages paradigm is documented here: https://rstudio.github.io/learnr/publishing.html#r-package*

I think we are both on the same page, so to speak :smiley_cat:

**Pull request merged**